src/main/java/alfred/action/Storage.java

```diff
@@ -0,0 +1,158 @@
+     *
+     * @throws FileNotFoundException if a TaskFile could not be found
+     */
+    public TaskList fileToTaskList() throws FileNotFoundException {
+        Scanner s = new Scanner(taskFile); // create a Scanner using the File as the source
+        String current;
+        TaskList tasks = new TaskList();
+        while (s.hasNext()) {
+            current = s.nextLine();
+            if (current.contains(Commands.COMMAND_TODO)) {
+                String[] descriptions = current.split(Commands.COMMAND_TODO);
+                Task newTask = new ToDo(Commands.COMMAND_TODO + descriptions[1],
+                        current.contains("[X]"));
+                tasks.addTasks(newTask);
+            } else if (current.contains(Commands.COMMAND_EVENT)) {
+                String[] descriptions = current.split(Commands.COMMAND_EVENT);
+                String openBracket = "\\(";
+                String closeBracket = "\\)";
+                String[] newDescriptions = descriptions[1].split(openBracket);
+                String date = newDescriptions[1].split(closeBracket)[0];
+                String[] dateSplits = date.split("at: ")[1].split(" ");
+                date = dateSplits[0] + " " + dateSplits[1] + " " + dateSplits[2];
+                String time = dateSplits[3] + dateSplits[4];
+                DateTimeFormatter formatterDate = DateTimeFormatter.ofPattern("MMM d yyyy");
+                DateTimeFormatter formatterTime = DateTimeFormatter.ofPattern("hh:mma");
+                LocalDate newDate = LocalDate.parse(date, formatterDate);
+                LocalTime newTime = LocalTime.parse(time, formatterTime);
+                Task newTask = new Event(Commands.COMMAND_EVENT + newDescriptions[0],
+                        newDate, newTime, current.contains("[X]"));
+                tasks.addTasks(newTask);
+            } else {
+                String[] descriptions = current.split(Commands.COMMAND_DEADLINE);
+                String openBracket = "\\(";
+                String closeBracket = "\\)";
+                String[] newDescriptions = descriptions[1].split(openBracket);
+                String date = newDescriptions[1].split(closeBracket)[0];
+                String[] dateSplits = date.split("by: ")[1].split(" ");
+                date = dateSplits[0] + " " + dateSplits[1] + " " + dateSplits[2];
+                String time = dateSplits[3] + dateSplits[4];
+                DateTimeFormatter formatterDate = DateTimeFormatter.ofPattern("MMM d yyyy");
+                DateTimeFormatter formatterTime = DateTimeFormatter.ofPattern("hh:mma");
+                LocalDate newDate = LocalDate.parse(date, formatterDate);
+                LocalTime newTime = LocalTime.parse(time, formatterTime);
+                Task newTask = new Deadline(Commands.COMMAND_DEADLINE + newDescriptions[0],
+                        newDate, newTime, current.contains("[X]"));
+                tasks.addTasks(newTask);
+            }
```

Comment:
Could there be line breaks to breakup the chunk of text to make it more readable?